src/main/java/ev3dev/utils/DataChannelRewriter.java

```diff
@@ -0,0 +1,75 @@
+package ev3dev.utils;
+
+import java.io.Closeable;
+import java.io.IOException;
+import java.nio.ByteBuffer;
+import java.nio.channels.FileChannel;
+import java.nio.charset.StandardCharsets;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.nio.file.StandardOpenOption;
+
+/**
+ * Writer of streams that can rewrite the same channel for structured data of
+ * known length. The focus of this class is on performance.
+ *
+ * @author David Walend
+ */
+public class DataChannelRewriter implements Closeable {
+
+    private final Path path;
+    private final ByteBuffer byteBuffer;
+    private final FileChannel channel;
+
+    /**
+     * Create a DataChannelRewriter for path with a bufferLength byte buffer
+     *
+     * @param path path to the file to reread
+     * @param bufferLength length of the buffer to hold the structure
+     */
+    public DataChannelRewriter(Path path, int bufferLength) {
+        this.path = path;
+        this.byteBuffer = ByteBuffer.allocate(bufferLength);
+        try {
+            this.channel = FileChannel.open(path, StandardOpenOption.WRITE);
```

Comment:
`StandardOpenOption.CREATE` might be needed by some unit tests in the future if it's not enabled by default already.

Reply:
It's not enabled by default. I had to use a different codepath to create the files in the test. (It does give a very nice message if it can't create a file.)

I'd like to leave it out until we actually need it. Are there any cases in ev3dev where we'd want DataChannelRewriter to create a new file - to control something in ev3dev? 

We'd also want to rename the class - it wouldn't be rewriting a new file.

Reply:
Not in ev3dev - it should be fine there. However, some sensor unit tests may be not be creating all attributes that the class uses. I agree that it might be better to just fix the tests instead of changing the code here.

Reply:
I really need to get some background on how the existing tests work. It's not what I thought I would find - via a properties change to set up a file system that looks like the ev3dev's, and I'm having trouble seeing through the varied test styles into the setup code.